etc/inc/upgrade_config.inc

```diff
@@ -2976,10 +3001,10 @@ function upgrade_083_to_084() {
 			$config['hasync'] = $config['installedpackages']['carpsettings']['config'][0];
 			unset($config['installedpackages']['carpsettings']);
 		}
-		if (empty($config['installedpackages']['carpsettings'])) {
+		if (empty($config['installedpackages']['carpsettings']) && isset($config['installedpackages']['carpsettings'])) {
 			unset($config['installedpackages']['carpsettings']);
 		}
-		if (empty($config['installedpackages'])) {
+		if (empty($config['installedpackages']) && isset($config['installedpackages'])) {
```

Comment:
var_dump(empty($config['installedpackages']));
bool(false)
var_dump(empty($config['crud']));
bool(true)

empty() on something that is not set returns true, just the same as if it is set to the blank string, or an empty array, or the number 0.
So just empty()=true does not mean it is set.